components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/GatewayPolicyDeployer.java

```diff
@@ -0,0 +1,191 @@
+                    APIGatewayAdmin apiGatewayAdmin = new APIGatewayAdmin();
+                    MessageContext.setCurrentMessageContext(
+                            org.wso2.carbon.apimgt.gateway.utils.GatewayUtils.createAxis2MessageContext());
+                    PrivilegedCarbonContext.startTenantFlow();
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+                    List<String> gatewayRuntimeArtifacts = ServiceReferenceHolder.getInstance().getArtifactRetriever()
+                            .retrieveGatewayPolicyArtifacts(encodedString, tenantDomain);
+                    if (gatewayRuntimeArtifacts.size() == 0) {
+                        return true;
+                    }
+                    for (String runtimeArtifact : gatewayRuntimeArtifacts) {
+                        try {
+                            if (StringUtils.isNotEmpty(runtimeArtifact)) {
+                                GatewayPolicyDTO gatewayPolicyDTO = new Gson().fromJson(runtimeArtifact,
+                                        GatewayPolicyDTO.class);
+                                apiGatewayAdmin.deployGatewayPolicy(gatewayPolicyDTO);
+                            }
+                        } catch (AxisFault axisFault) {
+                            log.error(
+                                    "Error in deploying gateway policy artifacts to the Gateway during server startup ",
+                                    axisFault);
+                        }
+                    }
+                    result = true;
+                } catch (ArtifactSynchronizerException | AxisFault e) {
+                    String msg = "Error deploying policy artifacts to the Gateway ";
+                    log.error(msg, e);
+                    return false;
+                } finally {
+                    MessageContext.destroyCurrentMessageContext();
+                    PrivilegedCarbonContext.endTenantFlow();
+                }
+            } else {
+                String msg = "Artifact retriever not found";
+                log.error(msg);
+                throw new ArtifactSynchronizerException(msg);
+            }
+        }
+        return result;
+    }
+
+    private GatewayPolicyDTO retrieveGatewayPolicyArtifact(String policyMappingUUUID)
+            throws ArtifactSynchronizerException {
+
+        GatewayPolicyDTO result;
+
+        if (artifactRetriever != null) {
+            try {
+                String gatewayRuntimeArtifact = artifactRetriever.retrieveGatewayPolicyArtifacts(policyMappingUUUID);
+                if (StringUtils.isNotEmpty(gatewayRuntimeArtifact)) {
+                    result = new Gson().fromJson(gatewayRuntimeArtifact, GatewayPolicyDTO.class);
+                } else {
+                    String msg = "Error retrieving artifacts for policy mapping UUID " + policyMappingUUUID +
```

Comment:
as per wso2 style guidelines, the operators should go to the start of the next line if statement expands to multiple lines.
```suggestion
                    String msg = "Error retrieving artifacts for policy mapping UUID " + policyMappingUUUID 
                               + ". Storage returned null";
```

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/1136a795360198388c6b4bd97029c20b0f898832

Thanks and Regards